test/zhr_devs/submissions/read_models/submissions_test.exs

```diff
@@ -0,0 +1,45 @@
+defmodule ZhrDevs.Submissions.ReadModels.SubmissionsTest do
+  use ExUnit.Case, async: true
+
+  alias ZhrDevs.Submissions.Events.SolutionSubmitted
+  alias ZhrDevs.Submissions.ReadModels.Submissions
+
+  import ZhrDevs.Fixtures
+
+  describe "increment_attemps/2" do
+    setup do
+      trigger_event = %SolutionSubmitted{
+        hashed_identity: generate_hashed_identity(),
+        technology: "elixir"
+      }
+
+      %{event: trigger_event}
+    end
+
+    test "with spawned identity - allows to increment without constraints", %{event: event} do
```

Comment:
I don't understand this at all. Are you showing how to write a bug?

Reply:
No, I was trying to demonstrate that we don't have any logic here besides incrementing. The aggregate is protecting us from incrementing above 2. I'm just documenting how the system behaves.

Or you think that we should put the logic that prevents increments above 2 here?

Reply:
It's very good that you made this point here, yeah.

I think that a correct way to model it in CRSQ methodology is by calling this an UpToCounter and instantiating it with 2 at this stage of our system.

As the system evolves, perhaps, we'll have another type of event like FlexiSolution* where we'll be making UpToCounter 10 in response to handling the event.

---

I guess my point is that this implementation a little bit violates the principle of "making illegal state unrepresentable".